PocketKit/Sources/PocketKit/Article/Highlights/ArticleComponentHighlights.swift

```diff
@@ -0,0 +1,249 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import DiffMatchPatch
+import SharedPocketKit
+import Sync
+import UIKit
+
+extension Array where Element == ArticleComponent {
+    /// Apply an array of patches to an array of `ArticleComponent`
+    /// - Parameter patches: the patches to apply
+    /// - Returns: the patched components
+    func highlighted(_ patches: [String]) -> [ArticleComponent] {
```

Comment:
One note here, is that this whole file given that its the meat, has a lot of conditionals, and has, lets call it, "fun" could prob do with some more comments about the purpose of every few lines.

Reply:
will do!